tools/prep_cc4_code.py

```diff
@@ -142,23 +148,60 @@ def handle_placeholders(self, content):
         for placeholder_pair in sorted(UpdateLicenseCode.placeholders):
             if self.has_placeholders(content, placeholder_pair):
                 self.log(
-                    "   Found placeholder: " + placeholder_pair + ", skipping",
+                    f"   Found placeholder: {placeholder_pair}, skipping",
                     "verbose",
                 )
             else:
                 start, end = UpdateLicenseCode.placeholders[placeholder_pair]
                 if placeholder_pair == "head":
                     target = "</head>"
-                    replacement = start + "\n" + end + "\n" + target
+                    replacement = f"{start}\n{end}\n{target}"
                 elif placeholder_pair == "header":
                     target = re.search("<body.*?>", content).group()
-                    replacement = target + "\n" + start + "\n" + end
+                    replacement = f"{target}\n{start}\n{end}"
                 elif placeholder_pair == "footer":
                     target = "</body>"
-                    replacement = start + "\n" + end + "\n" + target
+                    replacement = f"{start}\n{end}\n{target}"
                 elif placeholder_pair == "language-selector":
                     target = "<!-- Site Header End - DO NOT DELETE -->"
-                    replacement = target + "\n" + start + "\n" + end
+                    replacement = f"{target}\n{start}\n{end}"
+                elif placeholder_pair == "legalcode":
+                    re_pattern = re.compile(
+                        r"""
+                        ^\s*<div\ id="deed"
+                        .*
+                        ^\s*<li\ id="s8d">.*</li>\s*</ol>$
+                        (?=\s*<p\ class="shaded">)
+                        """,
+                        re.DOTALL | re.MULTILINE | re.VERBOSE,
+                    )
+                    target = re_pattern.search(content).group()
+                    replacement = f"\n{start}\n{target.strip()}\n{end}\n"
+                elif placeholder_pair == "language-footer":
+                    re_pattern = re.compile(
+                        r"""
```

Comment:
I think it's worth adding some comments documenting what these Regexes do (this one here and the one above).
They're not so small and it's not obviously clear at first glance what they do and what kind of HTML string they match with.